### PR TITLE
feat: auction liquidation impl

### PR DESCRIPTION
**Pull request by drbh**
## Security Checklist
WORK IN PROGRESS

- [ ] 1. Re-Entrancy
- [ ] 2. Arithmetic Over/Under Flows
- [ ] 3. Unexpected Ether
- [ ] 4. Delegatecall
- [ ] 5. Default Visibilities
- [ ] 6. Entropy Illusion
- [ ] 7. External Contract Referencing
- [ ] 8. Short Address/Parameter Attack (off chain)
- [ ] 9. Unchecked CALL Return Values
- [ ] 10. Race Conditions / Front Running
- [ ] 11. Denial Of Service (DOS)
- [ ] 12. Block Timestamp Manipulation
- [ ] 13. Constructors with Care
- [ ] 14. Uninitialized Storage Pointers
- [ ] 15. Floating Points and Precision
- [ ] 16. Tx.Origin Authentication
- [ ] 17. Address.isContract Re-Entrancy via Constructor

### ⚠️ NOTES ⚠️

Make sure to think about each of these exploits in this PR.

### CODE REVIEW

File: contracts/protocol/libraries/logic/AuctionLiquidationLogic.sol [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        require(
+            vars.healthFactor < THRESHOLD,
+            "Users health factor must be below the threshold"
+        );
+
+        /// TODO: settle debt from liquidation
+
+        // repay debt
+        // or supply -> wrapped eth gw
+        // also pay fee
```
Comment (drbh) [inline contracts/protocol/libraries/logic/AuctionLiquidationLogic.sol:145]: @WalidOfNow can you help point me to an example of how we might supply this back to the market?

File: contracts/protocol/libraries/logic/AuctionLiquidationLogic.sol [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            "Bid is too small, please increase"
+        );
+
+        require(
+            params.bidAmount > _auction.highestBid,
+            "Bid is not higher than existing highest bid"
+        );
+
+        // return previous amount to last highest bidder
+        payable(_auction.highestBidder).transfer(_auction.highestBid);
```
Comment (drbh) [inline contracts/protocol/libraries/logic/AuctionLiquidationLogic.sol:169]: @WalidOfNow can you help me determine the which contract the ETH should be moved in and out from? Maybe this should be on the pool?

File: contracts/protocol/pool/Pool.sol [hunk 3]
```diff
@@ -486,6 +487,63 @@ contract Pool is ReentrancyGuard, VersionedInitializable, PoolStorage, IPool {
...
+            })
+        );
+    }
+
+    function auctionLiquidationBid(
+        address token,
+        uint256 tokenId,
+        uint256 amount
+    ) external virtual {
+        AuctionLiquidationLogic.executeAuctionLiquidationBid(
```
Comment (drbh) [inline contracts/protocol/pool/Pool.sol:515]: @WalidOfNow should we move the ETH into the pool here?

File: contracts/protocol/libraries/logic/AuctionLiquidationLogic.sol (hunk 1 above)
Comment (WalidOfNow) [reply contracts/protocol/libraries/logic/AuctionLiquidationLogic.sol:145]: https://github.com/para-space/paraspace-core/blob/777a606a6a68154fe3882a83226d812b0d3b8d48/contracts/protocol/libraries/logic/LiquidationLogic.sol#L387

File: contracts/protocol/pool/Pool.sol (hunk 3 above)
Comment (WalidOfNow) [reply contracts/protocol/pool/Pool.sol:515]: the eth should has already been sent to the pool and can be checked in the `msg.value`

**Pull request closed without merging**